### PR TITLE
DOC: Update installation instructions

### LINKED ISSUE #3328: DOC: is install.txt outdated ?
Needs grammar/sentence check and check if information is outdated
while looking at #3281

### PR DESCRIPTION
**Pull request by bashtage**
Update installation instruction with more accurate information about
requirements and instructions for more recent version of Python

closes #3328
[skip ci]

### CODE REVIEW

**Comment by bashtage** [comment]
Reflects version in the updated Travis config

File: INSTALL.txt [hunk 1]
```diff
@@ -44,40 +43,36 @@ X-12-ARIMA or X-13ARIMA-SEATS
...
-------------
-
-To get the latest release using easy_install you need setuptools (easy_install):
-
-    http://peak.telecommunity.com/DevCenter/EasyInstall
+Installing
+----------
 
-Then you can do (with proper permissions):
+To get the latest release using git
```
Comment (TomAugspurger) [inline INSTALL.txt]: latest release using *pip* (not git). You might want to recommend the new `--upgrade-strategy="only-if-needed"` option for recent versions of pip.

File: docs/source/install.rst [hunk 2]
```diff
@@ -8,11 +8,12 @@ Installation
...
-To obtain the latest released version of statsmodels using `setuptools <https://pypi.python.org/pypi/setuptools>`__::
+To obtain the latest released version of statsmodels using pip
 
     pip install -U statsmodels
 
-Or follow `this link to our PyPI page <https://pypi.python.org/pypi/statsmodels>`__.
+Or follow `this link to our PyPI page <https://pypi.python.org/pypi/statsmodels>`__, download
+the wheel or source and install.
 
 Statsmodels is also available in `Anaconda <https://www.continuum.io/downloads>`__::
```
Comment (TomAugspurger) [inline docs/source/install.rst]: Better to point to conda forge I think: `conda install -c conda-forge statsmodels`.

**Comment by bashtage** [comment]
I have made suggestions along these lines.

**Comment by josef-pkt** [comment]
I'm merging this into master, but the python and numpy minimum versions are too high for 0.8

**Pull request merged**